moodle/moodle
# Add the ability to disable features supported by bigbluebutton disabledFeatures API.

**Background**
In 2.5 release of BigBlueButton, it had added disabledFeatures API which allows to disable a bunch of features in a particular meeting. Refer to below to learn more
https://docs.bigbluebutton.org/development/api/ 

This change adds the ability to disable/enable the same in a BigBlueButton activity in a Moodle course.

**Testing instructions**

	Go to Admin > Plugin > Activities, and enable BigBlueButton
	Click Settings and accept BigBlueButton terms of use
	Go to Admin > Plugin > Activity modules and click BigBlueButton
	Go to section 'Configuration for disabled features settings'
	Tick the below checkboxes
		bigbluebuttonbn_breakoutrooms_editable
		bigbluebuttonbn_captions_editable
		bigbluebuttonbn_chat_editable
		bigbluebuttonbn_downloadpresentationwithannotations_editable
		bigbluebuttonbn_snapshotofcurrentslide_editable
		bigbluebuttonbn_snapshotofcurrentslide_editable
		bigbluebuttonbn_externalvideos_editable
		bigbluebuttonbn_importpresentationwithannotationsfrombreakoutrooms_editable
		bigbluebuttonbn_importsharednotesfrombreakoutrooms_editable
		bigbluebuttonbn_layouts_editable
		bigbluebuttonbn_learningdashboard_editable
		bigbluebuttonbn_polls_editable
		bigbluebuttonbn_screenshare_editable
		bigbluebuttonbn_sharednotes_editable
		bigbluebuttonbn_virtualbackgrounds_editable
		bigbluebuttonbn_customvirtualbackgrounds_editable
		bigbluebuttonbn_presentationdf_editable
		bigbluebuttonbn_cameraascontent_editable
		bigbluebuttonbn_timer_editable
	Go to Admin > Development > Make a test course
	Select 'size of course' XS
	Course short name 'TestBBB'		
	Course full name 'TestBBB'
	Click 'Edit mode' and add activity > BigBlueButton
	Join this BigBlueButton session and see that the above settings are all available
	Go back to BigBlueButton activity and click 'Room settings' and tick each of the above settings
	Join this BigBlueButton and see that NONE of the above settings are available.	

_Note:_
All the features can be enabled/disabled except the below:
- liveTranscription (not supported)
- disable presentation 
-- cannot work because of the name is clashing and we need to work on this separately.

Regards,
Sumaiya
